Update Route4MeManager.cs

r4m-oleg opened the pull request:
Changes were made in order to improve the parsing of the server response.
Particularly, take into account received data of the type HttpConnectionResponseContent, as well as error messages.

r4m-oleg committed:
Update Route4MeManager.cs

pwnded commented on route4me-csharp-sdk/Route4MeSDKLibrary/Route4MeManager.cs:
```diff
@@ -6833,10 +6833,37 @@ public TerritoryZone UpdateTerritory(TerritoryZoneParameters territoryParameters
-                                    streamTask.Wait();
-
-                                    errorResponse = streamTask.Result.ReadObject<ErrorResponse>();
+                                    errorResponse = default; 
```
there is change for GetJsonObjectFromAPI, but there is no change for GetJsonObjectFromAPIAsync

Also GetXmlObjectFromAPI/GetXmlObjectFromAPIAsync is not considered

pwnded replied:
is it possible to add unit test to see the problem? Like having a red unit test without a fix, and green unit test with fix?

r4m-oleg committed:
Update Route4MeManager.cs

r4m-oleg committed:
update Route4MeSDKUnitTest project

r4m-oleg commented:
As for GetJsonObjectFromAPIAsync and GetXmlObjectFromAPIAsync, they do not need modification because they are asynchronous and therefore do not have problems similar to those mentioned above.

As for GetXmlObjectFromAPI, it is created for Geocoding, GeocodingAsync (Old version, still used by some users). It does not need modification, but I anyway made it for similarity.

Also, I modified some tests and added test data for Advanced Routing tests.

pwnded commented on route4me-csharp-sdk/Route4MeSDKLibrary/Route4MeManager.cs:
```diff
@@ -5,6 +5,7 @@
 using System.Net;
 using System.Net.Http;
 using System.Net.Http.Headers;
+using System.Security.Cryptography;
```
why do we need Cryptography here?

r4m-oleg replied:
We do not need, it remained from old case. We can remove it.

pwnded commented on route4me-csharp-sdk/Route4MeSDKLibrary/Route4MeManager.cs:
```diff
@@ -6624,19 +6626,19 @@ public TerritoryZone UpdateTerritory(TerritoryZoneParameters territoryParameters
                                 response.StatusCode == HttpStatusCode.RedirectMethod)
                             {
-                                var streamTask = await response.Content.ReadAsStreamAsync().ConfigureAwait(false);
+                                    var streamTask = await response.Content.ReadAsStreamAsync().ConfigureAwait(false);
```
is there any change or it is unintended formatting change?

r4m-oleg replied:
I first changed it, but then I saw that it did not make sense, so I restored the code, and unintended formatting happened.

pwnded commented:
> As for GetJsonObjectFromAPIAsync and GetXmlObjectFromAPIAsync, they do not need modification because they are asynchronous and therefore do not have problems similar to those mentioned above.
> 
> As for GetXmlObjectFromAPI, it is created for Geocoding, GeocodingAsync (Old version, still used by some users). It does not need modification, but I anyway made it for similarity.
> 
> Also, I modified some tests and added test data for Advanced Routing tests.



> 



> As for GetJsonObjectFromAPIAsync and GetXmlObjectFromAPIAsync, they do not need modification because they are asynchronous and therefore do not have problems similar to those mentioned above.
> 
> As for GetXmlObjectFromAPI, it is created for Geocoding, GeocodingAsync (Old version, still used by some users). It does not need modification, but I anyway made it for similarity.
> 
> Also, I modified some tests and added test data for Advanced Routing tests.

why async API does not have such issue? Async is just about the threading. But you change is about dealing with HTTP response, which is the same for sync and async.

r4m-oleg commented:
> 

because async variant does not contain casting to the type.
